### PR TITLE
Add a trivial review label

### LINKED ISSUE #34: Add support for label to merge trivial changes even if no reviews have been approved
For trivial small changes we've discussed adding support for having an opt-in label on them that would enable the octobors to skip waiting for approvals for pending reviewers and merge as soon as builds pass (and there are no "required changes").

This would greatly speed up PR merge flow by explicitly opting out of pre-merge reviews for some changes

### PR DESCRIPTION
**Pull request by bnjbvr**
This adds a new trivial review label to the configuration, which if it is present, makes it so that the PR doesn't block on reviews. Other preconditions may still prevail (e.g. CI must pass), but the requirement on reviewers in particular is relaxed.

Fixes #34.

### CODE REVIEW

**Comment by bnjbvr** [comment]
See it in action there: https://github.com/bnjbvr/octotest/pull/2

**Comment by bnjbvr** [comment]
See it in action there: https://github.com/bnjbvr/octotest/pull/2

**Review by Jake-Shadle — APPROVED** [review]

File: config/test.toml [hunk 1]
```diff
@@ -1,9 +1,10 @@
 owner = "bnjbvr"
-dry_run = true
+dry_run = false
```
Comment (davidpdrsn) [inline config/test.toml:2]: Was this change intentional? I don't where this config is used but feel dangerous to change 😬

File: src/process/tests.rs [hunk 2]
```diff
@@ -84,6 +87,28 @@ async fn merge_blocked_by_label() {
...
+    analyzer.reviews = RemoteData::Local(vec![review("1", ReviewState::ChangesRequested)]);
+
+    assert_eq!(
+        analyzer.required_actions().await.unwrap(),
+        *Actions::noop()
+            .set_merge(true)
+            .set_label("reviewed", Presence::Absent)
+            .set_label("ci-passed", Presence::Present)
+            .set_label("needs-description", Presence::Absent)
+    );
```
Comment (davidpdrsn) [inline src/process/tests.rs:109]: Does this mean if someone requests changes on a trivial pr it will still be merged. That doesn't feel right to me 🤔

File: src/process/tests.rs [hunk 3]
```diff
@@ -141,26 +166,25 @@ async fn no_description_none_required_pr_actions() {
...
-            assert_eq!(
-                analyzer.required_actions().await.unwrap(),
-                *Actions::noop()
-                    .set_merge($approved)
-                    .set_label("ci-passed", Presence::Present)
-                    .set_label("needs-description", Presence::Absent)
-            );
-        }};
+
+    async fn assert_approved(approved: bool, cases: Vec<Review>) {
```
Comment (davidpdrsn) [inline src/process/tests.rs:170]: This should make the backtrace still point to assertion

```suggestion
    #[track_caller]
    async fn assert_approved(approved: bool, cases: Vec<Review>) {
```

**Review by davidpdrsn — COMMENTED** [review]
A couple of drive by questions. Thanks for working on this!

File: src/process/tests.rs (hunk 2 above)
Comment (bnjbvr) [reply src/process/tests.rs:109]: Ah yeah, we could make it so that if somebody had the time to make a review, then the PR would not get merged 👍

File: config/test.toml (hunk 1 above)
Comment (bnjbvr) [reply config/test.toml:2]: Yeah this is my personal testing configuration on https://github.com/bnjbvr/octotest, a test repo dedicated for that, so it's fiiiiine :stuck_out_tongue:

**Review by davidpdrsn — APPROVED** [review]

File: config/example.toml [hunk 4]
```diff
@@ -27,6 +27,10 @@ ci_passed_label = "ci-passed"
 # have approved.
 reviewed_label = "reviewed"
 
+# Optional: Label that can be manually added to PRs to not block on reviews, for trivial changes.
+# If there's also a `block_merge_label` set, it has priority over this label being set.
+trivial_review_label = "trivial"
```
Comment (repi) [inline config/example.toml:32]: Can we call it this instead, think it is more explicit what the operation does and it sounds a bit scarier than a "trivial" change so devs hopefully take a bit more care into putting the label on it.    in our internal repos we will likely call this `s: skip review`

can we also have it not be active by default by not having it set? as we likely want to have it disabled by default in all repos using octobors and have them opt into it. one likely also want to only use it for private repos with trusted devs, rather than any open source repos

```suggestion
skip_review_label = ""
```

**Review by repi — CHANGES_REQUESTED** [review]
awesome! one comment on naming & default.

File: config/example.toml (hunk 4 above)
Comment (bnjbvr) [reply config/example.toml:32]: > Can we call it this instead, think it is more explicit what the operation does and it sounds a bit scarier than a "trivial" change so devs hopefully take a bit more care into putting the label on it. in our internal repos we will likely call this s: skip review

Happy to rename this, but according to the _specification and implemented behavior_: this will _not_ ignore a review that requested changes, so to some extent it does not override existing reviews. So "skip" didn't quite feel right to me. I don't have a very strong opinion, so if people preferred it we can use that instead (or anything else we'd come up with and agree upon).

> can we also have it not be active by default by not having it set?

I really hope that people are tweaking the values in the example.toml file before using it :sweat_smile: as it's rather intended to show all the possible options and their values. I'll comment this line in the file.

Comment (repi) [reply config/example.toml]: >  this will not ignore a review that requested changes, so to some extent it does not override existing reviews. So "skip" didn't quite feel right to me.

I guess technically it is "skip additional reviews" or "do not wait for reviews". think "skip" can work okay enough though and is more explicit than the previous "trivial" at least. with all labels we can have a longer description for what it means and does also. for the current `s: skip review` I have this as description (which we can update):

"This is a trivial change that will be automerged as soon as CI passes"

**Review by repi — APPROVED** [review]
